SeqAn Community Workflows
=========================

KNIME example workflows for NGS data analysis with SeqAn apps. Here we host and share community workflows.

Please read the howto at http://trac.seqan.de/wiki/HowTo/UseSeqAnNodesInKnime first about how to integrate
SeqAn nodes into KNIME and how to use or contribute workflows.

Workflows
---------

### Variant Calling ###
  <dl>
    <dt>[Variant Calling with SnpStore](variant_calling_snpstore_workflow)</dt>
    <dd>Two variant calling workflows based on the read mapper RazerS 3 and the variant caller SnpStore.</dd>
    <dt>[Variant Calling with mpileup](variant_calling_bwa_workflow)</dt>
    <dd>A variant calling workflow based on the read mapper BWA and samtools mpileup for variant calling.</dd>
  </dl>

### BS-Seq Analysis ###
  <dl>
    <dt>[BS-Seq Analysis with bs-tools](bs_seq_analysis_workflow)</dt>
    <dd>A workflow for BS-Seq data analysis, from bisulfite read mapping to SNP and methylation level calling at single-nucleotide resolution.</dd>
  </dl>

### Metagenomics ###
  <dl>
    <dt>[GASiC workflow](metagenomics_gasic_workflow)</dt>
    <dd>Genome Abundance Similarity Correction workflow.</dd>
  </dl>

### Benchmarking ###
  <dl>
    <dt>[Rabema Example Workflow](rabema_example_workflow)</dt>
    <dd>A workflow to benchmark the sensitivity of read mappers.</dd>
  </dl>
    
Contact
-------

If you have questions feel free to contact the SeqAn mailing list at seqan-dev@lists.fu-berlin.de
